### ISSUE TITLE
The method close() of PersistentBottomSheetController is not unblocking the screen if executes after dragging the modal bottom sheet down and then up. The screen stays blocked by half-transparent modal shadow.

### ISSUE THREAD

**Issue by ruan65** [flutter/flutter#47783]
## Steps to Reproduce

1. Issue description

The method close() of PersistentBottomSheetController is not unblocking the screen if executes after dragging the modal bottom sheet down and then up. The screen stays blocked by half-transparent modal shadow.

I am opening the scrollable bottom sheet like this:

`_controller = _scaffoldKey.currentState.showBottomSheet<void>(...);`

and then trying to close by pressing a button

`_controller.close();`

2. [demo video](https://drive.google.com/file/d/1F4QBr8b0QYtIPK07NwEtCPqLN3VuDuPm/view?usp=drivesdk)

3. The screen [code](https://gist.github.com/ruan65/33a78365f4e71037a0fb1340ddc363fd)

**Target Platform:** Android / iOS
**Target OS version/browser:** any
**Devices:** any

## Logs

```console
There are no exceptions
```

```console
flutter analyze does not have issues
```

flutter doctor -v

```console
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.14.6 18G1012, locale en-RU)
    • Flutter version 1.12.13+hotfix.5 at /Users/arudenko/soft/flutter
    • Framework revision 27321ebbad (2 weeks ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
    • Android SDK at /Users/arudenko/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.7.5

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] IntelliJ IDEA Ultimate Edition (version 2019.1.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 37.1.3
    • Dart plugin version 191.7830

[✓] VS Code (version 1.41.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.7.1

[✓] Connected device (1 available)
    • SM A505FN • R58M53PRDXA • android-arm64 • Android 9 (API 28)

• No issues found!


```

**Comment by sgehrman** [comment]
I have a similar problem.  
I'm using showBottomSheet and if I drag it down to dismiss, it leaves the modal barrier color on the scaffold.  showModalBottomSheet works better, but I don't want it modal.

**Comment by TahaTesser** [comment]
Hi @ruan65  @sgehrman 
Are you still experiencing this issue with latest `dev` channel?
If yes,  can you please provide your `flutter doctor -v` and `flutter run --verbose`? 
Also, to better address the issue, would be helpful if you could post a minimal code sample to reproduce the problem

@ruan65  code sample provided has missing code for `screenHeight` and `screenWidth`. **please provide complete sample**
Thank you

**Comment by iapicca** [comment]
Without additional information we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
